buildSrc/subprojects/buildquality/src/main/kotlin/org/gradle/gradlebuild/buildquality/classycle/ClassyclePlugin.kt

```diff
@@ -40,12 +40,14 @@ open class ClassyclePlugin : Plugin<Project> {
             val classycle by creating
             java.sourceSets.all {
                 val taskName = getTaskName(classycle.name, null)
-                val sourceSetTask = createTask(taskName, Classycle::class) {
-                    classesDirs = output.classesDirs
-                    excludePatterns.set(extension.excludePatterns)
-                    reportName = name
-                    reportDir = reporting.file(classycle.name)
-                    reportResourcesZip.set(extension.reportResourcesZip)
+                val sourceSetTask = project.tasks.create<Classycle>(
+                    taskName,
+                    output.classesDirs,
+                    extension.excludePatterns,
+                    name,
+                    reporting.file(classycle.name),
+                    extension.reportResourcesZip
+                ).apply {
                     dependsOn(output)
```

Comment:
❌ Looking at this again it seems like the `dependsOn` is unnecessary because `classesDirs` is already built by the `output`. https://github.com/gradle/gradle/blob/3ba92ea7d98eec46499e444bc0e3380c7bea24d1/subprojects/plugins/src/main/java/org/gradle/api/internal/tasks/DefaultSourceSetOutput.java#L48-L50

Could you try it out to make sure that compilation happens when you remove the explicit dependency?

Reply:
Looks like the same set of tasks run with or without the explicit `dependsOn` here, so I've removed it.

with `dependsOn`: https://e.grdev.net/s/rdb2xi74iozlc/timeline
without `dependsOn`: https://e.grdev.net/s/rub66a7rw2rds/timeline